README.md

```diff
@@ -34,6 +41,66 @@ Write a function to generate the fibonacci numbers
   - Write a function, `fibPairs`, so that when given an integer `n`, `fibPairs` returns a pair of integers representing the (n - 1)th and and nth fibonnacci numbers. You can take the -1st element of the fibonnaci sequence to be 0, and the 0th to be 1.
   - Write a function `fibonnaci` that maps an integer n to the nth fibonnacci number.
 
+## Recursion
+
+Spoiler warning -- Don't look in the Data.List module until you have working answers to this chapter's exercises.
+
+### Exercise (isPrefixOf)
+```haskell
+-- | The 'isPrefixOf' function takes two lists and returns 'True'
+-- iff the first list is a prefix of the second.
+isPrefixOf              :: (Eq a) => [a] -> [a] -> Bool
+```
+* Find a **recursive** implementation of `isPrefixOf`
+* Why do we enforce the typeclass constraint `(Eq a)`?
+
+### Exercise (intersperse)
+```haskell
+-- | The 'intersperse' function takes an element and a list and
+-- \`intersperses\' that element between the elements of the list.
+-- For example,
+--
+-- > intersperse ',' "abcde" == "a,b,c,d,e"
+```
+* Implement `intersperse` recursively
+
+### Exercise (transpose)
+```haskell
+-- | The 'transpose' function transposes the rows and columns of its argument.
+-- For example,
+--
+-- > transpose [[1,2,3],[4,5,6]] == [[1,4],[2,5],[3,6]]
+```
+* Implement `transpose`recursively
+
+## Higher Order Functions
+
+### Exercise (primes)
+
+Write functions that produce:
+* The 1st n primes
+* The nth prime
+* The first prime above a given integer
+
+Attempt the functions after reading **Maps and Filters**.
+For each following section, rewrite your functions with some new syntax you have learned.  Observe how the syntactic sugar melts away as you approach the end of this chapter.
+
+## Modules
+
+### Exercise (Making our own Modules)
+Put all your functions to handle primes into a module, exporting the three listed above.
+Load this module into ghci.
+
+## Making our own Types and Typeclasses
+
+### Exercise (anti-strings)
+
+In physics there re particles and antiparticles.  When a particle meets it's antiparticle counterpart they annihilate to nothing!
+Introducing anti-strings! E.g:
+```
+"disfunctional" ++ Antistring "fun" => "disctional"
```

Comment:
Eugh - do you really want to look inside the string (as opposed to annihilating the right-hand-side in the above example? What happens if you don't find "fun" inside the string you add to?

Also `(++) :: [a] -> [a] -> [a]`, whereas you need something else here. Would be worth writing down a few more examples of expected behaviour, and checking the exercise is possible in the framework you lay out (is it?).

As a general principle, it's worth noting that Haskell's typeclasses have a strong algebraic flavour. Strings form something called a `Monoid` under `(++)` - it looks like you are trying to make a `Group`.

Reply:
Yeah I haven't really thought this through properly yet.  I'll make it less specific and open-ended. Perhaps put it under a later chapter.